kommunicate/src/main/java/io/kommunicate/async/KmConversationCreateTask.java

```diff
@@ -0,0 +1,73 @@
+package io.kommunicate.async;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import com.applozic.mobicomkit.api.people.ChannelInfo;
+import com.applozic.mobicomkit.channel.service.ChannelService;
+import com.applozic.mobicomkit.feed.ChannelFeedApiResponse;
+
+import java.lang.ref.WeakReference;
+
+import io.kommunicate.callbacks.KMStartChatHandler;
+import io.kommunicate.callbacks.KmStartConversationHandler;
+
+public class KmConversationCreateTask extends AsyncTask<Void, Void, ChannelFeedApiResponse> {
+    private WeakReference<Context> context;
+    private ChannelService channelService;
+    private ChannelInfo channelInfo;
+    private KmStartConversationHandler startConversationHandler;
+    private KMStartChatHandler startChatHandler;
+
+
+    public KmConversationCreateTask(Context context, ChannelInfo channelInfo, KmStartConversationHandler startConversationHandler) {
+        this.context = new WeakReference<>(context);
+        this.startConversationHandler = startConversationHandler;
+        this.channelInfo = channelInfo;
+        this.channelService = ChannelService.getInstance(context);
+    }
+
+    public KmConversationCreateTask(Context context, ChannelInfo channelInfo, KMStartChatHandler startChatHandler) {
+        this.context = new WeakReference<>(context);
+        this.startChatHandler = startChatHandler;
+        this.channelInfo = channelInfo;
+        this.channelService = ChannelService.getInstance(context);
+    }
+
+    @Override
+    protected ChannelFeedApiResponse doInBackground(Void... voids) {
+        if (channelInfo != null) {
+            return channelService.createChannelWithResponse(channelInfo);
+        }
+        return null;
+    }
+
+    @Override
+    protected void onPostExecute(ChannelFeedApiResponse channelFeedApiResponse) {
+        super.onPostExecute(channelFeedApiResponse);
+        if (channelFeedApiResponse != null) {
+            if (channelFeedApiResponse.isSuccess()) {
+                if (startConversationHandler != null) {
+                    startConversationHandler.onSuccess(channelService.getChannel(channelFeedApiResponse.getResponse()), context.get());
+                }
+                if (startChatHandler != null) {
+                    startChatHandler.onSuccess(channelService.getChannel(channelFeedApiResponse.getResponse()), context.get());
+                }
+            } else {
+                if (startConversationHandler != null) {
```

Comment:
@reytum this looks same to the code block in another else. is it possible to put it in some variable then have only 1 common code. 
  if (startConversationHandler != null) {
                startConversationHandler.onFailure(VARIABLE, context.get());
            }
            if (startChatHandler != null) {
                startChatHandler.onFailure(null, context.get());
            }

Reply:
Here we need to trigger both the code blocks based on whether the response is null or the response is error. Did you mean that we need to move the complete failure code block to another method and then call the method in both the places?

Reply:
yes

Reply:
Done